snapcraft/plugins/gradle.py

```diff
@@ -92,7 +105,8 @@ def build(self):
             gradle_cmd = ['gradle']
         self.run(gradle_cmd +
                  self._get_proxy_options() +
-                 self.options.gradle_options + ['jar'])
+                 self.options.gradle_options +
+				 self.options.gradle_tasks)
```

Comment:
I'm all for this change in principle, as gradle isn't limited to the `jar` tasks, or even the `java` plugin (there's a `cpp` one, for example). However, now that we've opened up the ability to use gradle for other things, we still have the following logic that looks for .jar or .war files and bails out if it doesn't find them. If we open up this capability, we probably need to make that more generic as well. Thoughts?

Reply:
so I couple of issues. 1) I'm new to gradle 2) I'm not a python programmer.

I agree that we should extend this to support other outputs, however I've not sufficient knowledge of gradle/python to propose a patch that is likely to do the required job.

So two options:
1) we take the pull request as is as it adds useful functionality and does it in a way that would allow for a future update to support other output targets

2) we find a second contributor that can provide some advice on what's required to make this more generic to other output targets.

I'm inclined to go for 1) as it makes the new functionality available now and I think its sufficiently low risk that it will make future enhancements break backwards compatibility.

Reply:
As to tests can you point me in the right direction as to how to create a test/where to place a test for this?

Reply:
OK, so I found the test cases, but in looking at the code I think its going to be well beyond my ability with python to build a coherent test case for this.
I'm using it in production so it does work, but that doesn't help for future maintainability.

Reply:
OK, so wasn't as hard I as i thought.
I've created some unit tests but I'm unable to run them the the dev setup guide is a bit vauge.

I'm getting the following error, what do I need to install to run the unit tests?

Google didn't turn up anything useful.


/runtests.sh tests/unit
tests.unit (unittest.loader._FailedTest) ... ERROR

======================================================================
ERROR: tests.unit (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.unit
Traceback (most recent call last):
  File "/usr/lib/python3.6/unittest/loader.py", line 462, in _find_test_path
    package = self._get_module_from_name(name)
  File "/usr/lib/python3.6/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/home/bsutton/git/snapcore/snapcraft/tests/unit/__init__.py", line 23, in <module>
    import fixtures
ModuleNotFoundError: No module named 'fixtures'